File: src/main/java/me/elsiff/morefish/condition/EnchantmentCondition.java
package me.elsiff.morefish.condition;

import org.bukkit.enchantments.Enchantment;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;

public class EnchantmentCondition implements Condition {
    private Enchantment enchantment;
    private int level;

    public EnchantmentCondition(Enchantment enchantment, int level) {
        this.enchantment = enchantment;
        this.level = level;
    }

    @Override
    public boolean isSatisfying(Player player) {
        ItemStack hand = player.getInventory().getItemInMainHand();
        return (hand != null && hand.containsEnchantment(enchantment) &&
                hand.getEnchantmentLevel(enchantment) >= level);
    }
}

File: src/main/kotlin/me/elsiff/morefish/fishing/condition/EnchantmentCondition.kt
package me.elsiff.morefish.fishing.condition

import me.elsiff.morefish.fishing.competition.FishingCompetition
import org.bukkit.enchantments.Enchantment
import org.bukkit.entity.Item
import org.bukkit.entity.Player

class EnchantmentCondition(
    private val enchantment: Enchantment,
    private val minLevel: Int
) : FishCondition {
    override fun check(
        caught: Item,
        fisher: Player,
        competition: FishingCompetition
    ): Boolean {
        val fishingRod = fisher.inventory.itemInMainHand
        return fishingRod.containsEnchantment(enchantment) && fishingRod.getEnchantmentLevel(enchantment) >= minLevel
    }
}